doc/null_safety/null_safe_migration.md

```diff
@@ -68,7 +116,108 @@ Below is a table of the possible options for prop nullability:
   - **Nullable**: If the prop is required, but can be explicitly set to `null`.
   - **Non-nullable**: If the prop is required and should never be set to `null`.
 
-#### Migrator tool bug
+If you aren't sure whether the prop should be nullable, or required, or both, this decision tree might help:
+
+```mermaid
+flowchart TD
+    Start[<strong>Should My Prop Be Required</strong>]==>HasDefault
+      HasDefault((Does the \nprop have a \ndefault value?))== Yes ==> Defaulted
+        Defaulted((Where is \nthe default?))-- defaultProps getter\n(Class Component) --> End_Defaulted1[/"Make it <strong>required</strong>\n<code>late SomeType propName;</code>"\]
```

Comment:
Hmm I wonder if we should add a note here about considering making them optional if they're mixed in in other packages if you're concerned about breakages? Thoughts?

Reply:
https://github.com/Workiva/over_react/pull/932/commits/e94e4253a2b8ddbfcc30c46cbdb6848acaf7d04e